chart/templates/dag-processor/dag-processor-deployment.yaml

```diff
@@ -206,12 +206,15 @@ spec:
           {{- if .Values.dagProcessor.logGroomerSidecar.args }}
           args: {{- tpl (toYaml .Values.dagProcessor.logGroomerSidecar.args) . | nindent 12 }}
           {{- end }}
-          {{- if .Values.dagProcessor.logGroomerSidecar.retentionDays }}
           env:
+          {{- if .Values.dagProcessor.logGroomerSidecar.retentionDays }}
             - name: AIRFLOW__LOG_RETENTION_DAYS
               value: "{{ .Values.dagProcessor.logGroomerSidecar.retentionDays }}"
+          {{- end }}
             - name: AIRFLOW_HOME
               value: "{{ .Values.airflowHome }}"
```

Comment:
This used to be that if retentionDays was not set, AIRFLOW_HOME would not be set. The new behavior always sets AIRFLOW_HOME. is that intentional? It seems to me like the new behavior would be correct, but it is a change from the old behavior. I think we should get an opinion from somebody on the airflow team.

Reply:
seems like it was a bug

Reply:
@dstandish When we submit this as a PR to apache/airflow, will they ask for any changes, or do you think this will be fine as it is?

Reply:
PR is here btw: https://github.com/apache/airflow/pull/46003/

Reply:
There is also this pr to add env to these containers: https://github.com/apache/airflow/pull/45483

Different approaches, but I think this one is better.

Not sure why this and the OSS one are different though. I agree - not setting home seems like a bug.

Reply:
Thanks! I just want to make sure we're not diverging from upstream airflow here.